docs/knowledgebase/advanced/try-runtime.md

```diff
@@ -0,0 +1,194 @@
+
+## How it works
+The `try-runtime` tool has its own implementation of externalities called [`remote_externalities`][remoteextern-rustdocs] which is just a builder wrapper around `TestExternalities` that uses
+a generic [key-value store](/docs/en/knowledgebase/advanced/storage) where data is [SCALE encoded](/docs/en/knowledgebase/advanced/codec).
+
+The diagram below illustrates the way externalities sits outside a compiled runtime as a means to capture 
+the storage of that runtime. 
+
+Storage externalities            |  Testing with externalities
+:-------------------------:|:-------------------------:
+![image](/docs/assets/advanced/try-runtime-ext-1.png)  |  ![image](/docs/assets/advanced/try-runtime-ext-2.png)
+
+With `remote_externalities`, developers can capture some chain state and run tests on it. Essentially, `RemoteExterrnalities` will populate a `TestExternalities` with a real chain's data. 
+
+In order to query state, `try-runtime` makes use of Substrate's RPCs, namely [`StateApi`][stateapi-rustdocs]. In particular:
+- [`storage`][stateapi-storage-rustdocs]: A method which returns a storage value under the given key.
+- [`storage_key_paged`][stateapi-storage-keys-paged-rustdocs]: A method which returns the keys with prefix with pagination support.
+
+
+## Usage
+
+The most common use case for `try-runtime` is with storage migrations and runtime upgrades. 
+
+> **Tip:** Combine `try-runtime` with `fork-off-substrate` to test your chain before production. Use `try-runtime` to test your 
+> chain's migration and its pre and post states. Then, use [`fork-off-substrate`][fork-off-gh] if you want to check that block 
+> production continues fine after the migration, and do some other arbitrary testing. 
+
+### Calling into hooks from `OnRuntimeUpgrade`
+By default, there are two ways of defining a runtime upgrade in the runtime. The `OnRuntimeUpgrade` trait provides the [different methods][onruntimeupgrade-method-rustdocs] to achieve this.
+
+- **From inside a runtime**. For example:
+    ```rust
+    struct Custom;
+    impl OnRuntimeUpgrade for Custom {
+        fn on_runtime_upgrade() -> Weight { }
+    }
+    ```
+
+- **From inside a pallet**. For example:
+    ```rust
+    #[pallet::hooks]
+    fn on_runtime_upgrade() -> Weight { }
+    ```
+
+To call into these hooks with `try-runtime`, a cargo `feature` is used. This requires specifying a `features` 
```

Comment:
This is false. To call into these hooks, you don't need a feature. To call into the additional hooks (e.g. `pre_upgrade`) you need the feature.